tests/rptest/tests/datalake/compaction_gaps_test.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2024 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from time import time
+from rptest.clients.kafka_cat import KafkaCat
+from rptest.clients.types import TopicSpec
+from rptest.services.kgo_verifier_services import KgoVerifierProducer
+from rptest.tests.datalake.datalake_services import DatalakeServices
+from rptest.tests.datalake.query_engine_base import QueryEngineType
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.redpanda import SISettings
+from rptest.tests.datalake.utils import supported_storage_types
+from ducktape.mark import matrix
+from ducktape.utils.util import wait_until
+from rptest.services.cluster import cluster
+from rptest.utils.mode_checks import skip_debug_mode
+
+
+class CompactionGapsTest(RedpandaTest):
+    def __init__(self, test_ctx, *args, **kwargs):
+        super(CompactionGapsTest, self).__init__(
+            test_ctx,
+            num_brokers=1,
+            si_settings=SISettings(test_context=test_ctx, fast_uploads=True),
+            extra_rp_conf={
+                "iceberg_enabled": "true",
+                "iceberg_catalog_commit_interval_ms": 5000,
+                "datalake_coordinator_snapshot_max_delay_secs": 10,
+                "log_compaction_interval_ms": 2000
+            },
+            *args,
+            **kwargs)
+        self.test_ctx = test_ctx
+        self.topic_name = "test"
+        self.segment_size = 5 * 1024 * 1024
```

Comment:
nit: could go even lower I think, especially since we're waiting for some number of segments